frontend/src/component/ui/Icon.tsx

```diff
@@ -0,0 +1,98 @@
+import React, { Component } from 'react';
+
+import './Icon.scss';
+
+export enum IconID {
+  Menu,
+  MenuOpen,
+  Close,
+  Search
+}
+
+interface IProps {
+  defaultIconID: IconID;
+  onClick?: () => void;
+}
+
+interface IStates {
+  iconID: IconID;
+}
+
+export class Icon extends Component<IProps, IStates> {
+  constructor(props: IProps) {
+    super(props);
+    this.state = {
+      iconID: props.defaultIconID
+    };
+  }
+
+  setIcon(iconID: IconID) {
+    this.setState({ iconID: iconID });
+  }
+
+  render() {
+    return (
+      <i className={'icon'} onClick={this.handleClick}>
+        {this.renderSVG()}
+      </i>
+    );
+  }
+
+  private handleClick = () => {
+    if (!this.props.onClick) {
+      return;
+    }
+    this.props.onClick();
+  };
+
+  private renderSVG() {
+    const { iconID } = this.state;
+
+    switch (iconID) {
+      case IconID.Close:
+        return (
+          <svg
+            className={'icon-close'}
+            xmlns="http://www.w3.org/2000/svg"
+            viewBox="0 0 24 24"
+          >
+            <path d="M0 0h24v24H0z" fill="none" />
+            <path d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z" />
+          </svg>
```

Comment:
I rethought about this. I think we should move the `SVGs` into `private renderClose` to avoid have a super long function.